### DISCUSSION TITLE
How can I override the Livy Connection in Airflow Livy Operator

### DISCUSSION THREAD

**Discussion by gowthamprabhus** [apache/airflow#26346] [Q&A] [answered]
I have dynamic Livy URLS for each Job. How can I override the livy_default connection and pass my Dynamic URL to Livy Operator ?

**Comment by potiuk** [comment, marked as answer]
If you have truly dynamic URL that you are not able to create a "pool" of connections - you will have to do it via custom operator/hook and override it. All operators and Hooks assume that there is a connection id that keeps host/authentication etc. so if you do not have a pool of connections, but truly dynamic (and random for example) URL, you will have to take the LivyHook's (and HTTP's Hook that Livy Hook is built on) code and write your own Hook to take host as parameter.

If you have pool of connections - then it's a bit easier, you define all the connections and name them "livy_conn_1", "livy_conn_2" etc. and then you can very simply extend LivyOperator and add conn_id to "template_fields" and use the regular Jinja templating.

There is potentially a way that you could use even truly dynamic URLs I think. You would have to create your own custom secret backend: https://airflow.apache.org/docs/apache-airflow/2.3.4/security/secrets/secrets-backend/index.html#roll-your-own-secrets-backend and your connection_id to template_fields in your custom operator and implement your secret backend to dynamically return the Connection based on the "id" that you will pass (and then you could use the variable part in the "conn_id" templated field in your DAG.
